src/vcpkg/base/files.cpp

```diff
@@ -1583,7 +1462,18 @@ namespace vcpkg
 
     bool has_invalid_chars_for_filesystem(const std::string& s)
     {
-        return std::regex_search(s, FILESYSTEM_INVALID_CHARACTERS_REGEX);
+        for (const char& c : s)
+        {
+            for (const char& invalid : {'[', '/', '\\', ':', '*', '"', '<', '>', '|', ']'})
+            {
+                if (c == invalid)
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
```

Comment:
```suggestion
        return s.find_first_of(R"([/\:*"<>|])") == std::string::npos;
```

Reply:
actually, even better:

```suggestion
        return s.find_first_of(FILESYSTEM_INVALID_CHARACTERS) == std::string::npos;
```